### ISSUE TITLE
Overwriting an existing project doesn't update the project's default "Save as" name.

### ISSUE THREAD

**Issue by Scoremonger** [Scirra/Construct-3-bugs#1060]
## Problem description

If you "Save as" and select an existing project and overwrite it, the next time you open that project and choose to "Save as", the project name won't have updated to reflect the new project name.

## Attach a .c3p

This can be reproduced with a new, empty project.

## Steps to reproduce

1. Save a project by choosing Menu -> Project -> Save as -> Cloud save. 
2. Type in a unique filename (i.e. "fred.c3p") and save.
3. Open another project ("Project1.c3p", for example) and Save as in the cloud, but this time select "fred.c3p" and overwrite it.
4. Close the project and open "fred.c3p".
5. Save as in the cloud, and notice the default save name.

## Observed result

The default save name is still whatever it was before being saved as fred.c3p (i.e. "Project1.c3p").

## Expected result

The default Save As name should be the same as the filename that was last used when saving the file.

## Affected browsers

* Chrome: yes

## System details

*If you see a crash report dialog, please copy and paste it to where it says "PASTE HERE" below. Otherwise please go to **Menu** -> **About** -> **Platform information** and paste that information there instead.*

<details><summary>View details</summary>

Platform information
Browser: Chrome
Browser version: 62.0.3202.94
Browser engine: Blink
Browser architecture: 64-bit
Context: webapp
Operating system: Windows
Operating system version: 10
Operating system architecture: 64-bit
Device type: desktop
Device pixel ratio: 1
Logical CPU cores: 8
Approx. device memory: (unavailable)
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.94 Safari/537.36
C3 release: r70.2
Language setting: en-US

Local storage
Storage quota (approx): 14 gb
Storage usage (approx): 182 mb (1.3%)
Persistant storage: Yes

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

This browser does not support pluralisation rules. Plural forms may not be correct.
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: yes
Vendor: Google Inc.
Renderer: ANGLE (Intel(R) HD Graphics 4000 Direct3D11 vs_5_0 ps_5_0)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 1024
Extensions:

EXT_color_buffer_float
EXT_disjoint_timer_query_webgl2
EXT_texture_filter_anisotropic
OES_texture_float_linear
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
Audio information
System sample rate: 48000 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

Ogg Opus (audio/ogg; codecs=opus)
WebM Opus (audio/webm; codecs=opus)
Ogg Vorbis (audio/ogg; codecs=vorbis)
WebM Vorbis (audio/webm; codecs=vorbis)
MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav; codecs=1)
Supported encode formats:

WebM Opus (audio/webm; codecs=opus)
Video information
Supported decode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
Ogg Theora (video/ogg; codecs=theora)
H.264 (video/mp4; codecs=avc1.42E01E)
Supported encode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)

</details>

**Comment by shortercode** [comment]
This is impressively specific... Just trying to get the replication down myself.  I'm presuming you close fred.c3p after step 2?

**Comment by shortercode** [comment]
I can't get this to happen, can you please try repeating your replication steps and adding in any missing steps for me? Given the weird combinations of opening closing files I expect I might be missing one or adding one in somewhere.

**Comment by Scoremonger** [comment]
I should mention I'm using One Drive, and I haven't tested this with other cloud locations. Let me start from scratch and get more detailed repro steps.

**Comment by Scoremonger** [comment]
Huh... the bug no longer seems to happen as I wrote it. Or maybe I made an error. But! There is still an inconsistency which hopefully these steps will demonstrate:

Open a project.
Save the project by choosing Menu -> Project -> Save as -> Cloud save.
Double-click an existing project to overwrite it with the current project.
**Without closing it**, save the current project again by choosing Menu -> Project -> Save as -> Cloud save.
The default filename will not be the file you just overwrote.

This is only the case if you double-click to overwrite the existing file - if you manually type in the same name as an existing file and overwrite, the next time you "save as", the proper default will be there.

**Comment by shortercode** [comment]
Ah yeah, you'll be glad to hear I can see this. I'll see if I can figure out whats going on.

**Comment by shortercode** [comment]
Okay I think I've got a fix for this. The file ID wasn't being updated when a project was "updated". Turns out it isn't normally a problem, because the value gets updated when the file is reopened and provided your saving to the same file the id doesn't change. So it's only when you save an open file to a different one, then try to save it again to the same service (google drive, browser, onedrive, etc. ) that it gets confused.
